### PR TITLE
revert the Promise.all() to a simple for loop to avoid any race condi…

### PR DESCRIPTION
**Pull request by thombergs**
…tions that we might experience.

### CODE REVIEW

File: src/github/middleware.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ export default (
 
 			if (await booleanFlag(BooleanFlags.MAINTENANCE_MODE, false, jiraHost)) {
 				context.log(`Maintenance mode ENABLED for jira host ${jiraHost} - Ignoring event of type ${webhookEvent}`);
-				return;
+				continue;
```
Comment (thombergs) [inline src/github/middleware.ts:131]: We could have kept these 3 `returns`, but it's cleaner and more refactoring-safe if we never return from within the loop.

**Review by KAbakumovAtl — APPROVED** [review]

File: src/github/middleware.ts (hunk 1 above)
Comment (KAbakumovAtl) [reply src/github/middleware.ts:131]: Shame on me, I didnt' check for other returns!

File: src/github/middleware.ts [hunk 2]
```diff
@@ -142,7 +142,7 @@ export default (
 					},
 					`Timing out at after ${context.timedout}ms`
 				);
-				return;
+				continue;
```
Comment (thombergs) [inline src/github/middleware.ts:145]: Actually, this return was plain wrong. After a timeout we should at least try to send the data to the next Jira. But it was always wrong...

File: src/github/middleware.ts (hunk 1 above)
Comment (KAbakumovAtl) [reply src/github/middleware.ts:131]: Let's extract the code within for loop to a separate function. I am still getting confused

**Review by mboudreau — APPROVED** [review]

**Pull request merged**